rc/filetype/markdown.kak

```diff
@@ -31,6 +31,15 @@ hook -group markdown-highlight global WinSetOption filetype=markdown %{
     hook -once -always window WinSetOption filetype=.* %{ remove-highlighter window/markdown }
 }
 
+declare-option \
+    -docstring "map of supported languages in fenced code blocks. Keys determine which languages are recognised. The value names the corresponding highligher, if empty the key is used instead." \
+    str-to-str-map markdown_supported_languages \
+    awk= c= cabal= clojure= coffee= cpp= crystal= css= cucumber= d= diff= dockerfile= elixir= erlang= fish= \
+    gas= go= haml= haskell= hs=haskell html= ini= java= javascript= js=javascript json= julia= kak=kakrc kickstart= \
```

Comment:
I initially wondered why the `str-to-str-map` instead of a `str-list` (and just keep our hack for `kakrc`).
The second commit reveals why: this allows to add filetype aliases like hs=haskell.
Prior discussion on that is in https://github.com/mawww/kakoune/pull/4592
where I thought it's ugly to add a new highlighter for each separate alias (there can be many)
but with your solution, users are responsible for adding the aliases
they need, so this seems okay.

Reply:
The decision to choose a map was motivated by the use of the `kakrc` module for blocks marked `kak`. But indeed, this allows for a much more general approach to managing recognised languages.